### ISSUE TITLE
[BUG] Edge function timeouts with long prompts

### ISSUE THREAD

**Issue by Ceiridge** [enricoros/big-agi#262]
**Describe the bug**
When prompting with at least 60.000 tokens on Claude 2.1, the following error is returned:

```
An error occurred with your deployment

EDGE_FUNCTION_INVOCATION_TIMEOUT
```

**Expected behavior**
An actual answer.

Is this related to your big-agi deployment on Vercel probably, so would self-hosting fix this?

**Comment by enricoros** [comment]
Yes, Vercel has a [25 seconds timeout](https://vercel.com/docs/functions/edge-functions/limitations#maximum-initial-response-time) on the first 'streaming byte' of an Edge function. Plus there's a queue (if you already have open requests, it could add up). 

big-AGI implements streaming as much as possible (e.g. chat/generate with stream=1) to prevent vercel from cutting the connections. However there are cases where upstream servers (e.g. Anthropic in this case) do not respond within 25 seconds and Vercel will time out.

Fix it with:
 - local deployment (from source or docker)
 - other cloud deployment with a larger timeout
 
Let me know if you are able to fix this.

**Comment by Ceiridge** [comment]
I won't deploy this, because I have found another solution for my use case.
But maybe you could add an option to disable streaming or send filler bytes to prevent a timeout.
